components/cell/jwks-server/crypto/resolver/key_resolver.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package resolver
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	b64 "encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"github.com/cellery-io/mesh-security/components/cell/jwks-server/resources"
+	"log"
+	"math/big"
+	"time"
+)
+
+var (
+	keyB  []byte
+	certB []byte
+)
+
+func KeyGenerator() (resources.JwksJson, error) {
+	keyB, certB, err := getKeyPair()
+	if err != nil {
+		return resources.JwksJson{}, err
+	}
+	jwksJson, err := KeyResolver(keyB, certB)
+	if err != nil {
+		return jwksJson, err
+	}
+	return jwksJson, nil
+}
+
+func KeyResolver(privateKeyStr []byte, certificateStr []byte) (resources.JwksJson, error) {
+	blockPriv, _ := pem.Decode([]byte(privateKeyStr))
+	// To get RSA Private key
+	key, err := x509.ParsePKCS1PrivateKey(blockPriv.Bytes)
+	if err != nil {
+		log.Printf("Error parsing private key. %s\n", err)
+		return resources.JwksJson{}, err
+	}
+
+	blockCert, _ := pem.Decode([]byte(certificateStr))
+	// To obtain a single certificate from the given ASN.1 DER data.
+	cert, err := x509.ParseCertificate(blockCert.Bytes)
+	if err != nil {
+		log.Printf("Error parsing certificate. %s\n", err)
+		return resources.JwksJson{}, err
+	}
+	log.Println("Decoded and parsed key and cert.")
+	if cert == nil {
+		log.Println("Unable to find the cert.")
```

Comment:
This log seems not descriptive enough. Would have been better if you say where the certs were searched in order to find out since we depend on multiple resources